### PR TITLE
ingress paths are now clickable

### PR DESCRIPTION
**Pull request by marcosdiez**
This is a copy of #7017
163737717-08e42793-e9e9-4ba5-b349-4a9390c59af4

-----------

sample YAML to test

```
kind: Ingress
apiVersion: networking.k8s.io/v1
metadata:
  name: kubernetes-dashboard-lb2
  namespace: kubernetes-dashboard
  annotations:
    kubernetes.io/ingress.class: nginx
    nginx.ingress.kubernetes.io/backend-protocol: HTTPS
spec:
  defaultBackend:
    service:
      name: test
      port:
        number: 80
  rules:
    - host: dashboard.mydomain.com
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: kubernetes-dashboard
                port:
                  number: 443
    - host: dashboard2.mydomain.com
      http:
        paths:
          - path: /mypath
            pathType: Prefix
            backend:
              service:
                name: kubernetes-dashboard
                port:
                  number: 443
    - host: dashboard2.mydomain.com
      http:
        paths:
          - path: /mypath/morepath
            pathType: Prefix
            backend:
              service:
                name: kubernetes-dashboard
                port:
                  number: 443
    - http:
        paths:
        - path: /testpath
          pathType: Prefix
          backend:
            service:
              name: test
              port:
                number: 80
    - http:
        paths:
          - path: /icons
            pathType: ImplementationSpecific
            backend:
              resource:
                apiGroup: k8s.example.com
                kind: StorageBucket
                name: icon-assets
    - host: "foo.bar.com"
      http:
        paths:
          - pathType: Prefix
            path: "/bar"
            backend:
              service:
                name: service1
                port:
                  number: 80
    - host: "*.foo.com"
      http:
        paths:
          - pathType: Prefix
            path: "/foo"
            backend:
              service:
                name: service2
                port:
                  number: 80
    - host: foo.bar.com
      http:
        paths:
          - path: /foo
            pathType: Prefix
            backend:
              service:
                name: service1
                port:
                  number: 4200
          - path: /bar
            pathType: Prefix
            backend:
              service:
                name: service2
                port:
                  number: 8080
```

### CODE REVIEW

File: modules/web/src/common/components/ingressrulelist/template.html [hunk 1]
```diff
@@ -46,7 +46,14 @@
         <mat-header-cell *matHeaderCellDef
                          i18n>Path </mat-header-cell>
         <mat-cell *matCellDef="let ingressRuleFlat">
-          {{ !!ingressRuleFlat?.path?.path ? ingressRuleFlat?.path?.path : '-' }}
+          <ng-container *ngIf="ingressRuleFlat.path.path">
+            <ng-container *ngIf="ingressRuleFlat.host">
```
Comment (maciaszczykm) [inline modules/web/src/common/components/ingressrulelist/template.html:50]: Can you merge these two into one `ingressRuleFlat.path.path && ingressRuleFlat.host`?

File: modules/web/src/common/components/ingressrulelist/template.html [hunk 2]
```diff
@@ -46,7 +46,14 @@
...
+              <a href="https://{{ ingressRuleFlat.host }}{{ ingressRuleFlat?.path?.path }}"
```
[the remark this answers is not in the document]
Comment (maciaszczykm) [reply modules/web/src/common/components/ingressrulelist/template.html:51]: There is no need to use `?.` as `*ngIf` above already checks that. I think it should be in the first changed line.

[the remark this answers is not in the document]
Comment (floreks) [reply modules/web/src/common/components/ingressrulelist/template.html:51]: I'd actually move this logic to the typescript and use a single method to handle path.

[the remark this answers is not in the document]
Comment (maciaszczykm) [reply modules/web/src/common/components/ingressrulelist/template.html:51]: Yeah, that would be the best.

[the remark this answers is not in the document]
Comment (marcosdiez) [reply modules/web/src/common/components/ingressrulelist/template.html]: I removed the `?`s.

I considered creating a typescript method (below), but I am not sure we want a method returning raw HTML.

I can change that if you guys prefer, though.

```
  createIngressLink(ingressRuleFlat: IngressRuleFlat): string {
    if (ingressRuleFlat.path.path) {
      if (ingressRuleFlat.host) {
        return '<a href="https://' + ingressRuleFlat.host + ingressRuleFlat.path.path + ' target="_blank">' + ingressRuleFlat.path.path + '</a>';
      } else {
        return ingressRuleFlat.path.path;
      }
    } else {
      return "-";
    }
  }
```

[the remark this answers is not in the document]
Comment (maciaszczykm) [reply modules/web/src/common/components/ingressrulelist/template.html]: Use one method for URL, second for label.

[the remark this answers is not in the document]
Comment (floreks) [reply modules/web/src/common/components/ingressrulelist/template.html]: It could also return a simple wrapper object with label and URL fields.

**Comment by maciaszczykm** [comment]
@marcosdiez Any updates?

**Comment by marcosdiez** [comment]
@maciaszczykm ready to merge!

File: modules/web/src/common/components/ingressrulelist/template.html [hunk 3]
```diff
@@ -46,7 +46,14 @@
         <mat-header-cell *matHeaderCellDef
                          i18n>Path </mat-header-cell>
         <mat-cell *matCellDef="let ingressRuleFlat">
-          {{ !!ingressRuleFlat?.path?.path ? ingressRuleFlat?.path?.path : '-' }}
+          <ng-container *ngIf="ingressRuleFlat.path.path">
+            <ng-container *ngIf="ingressRuleFlat.host">
+              <a href="https://{{ ingressRuleFlat.host }}{{ ingressRuleFlat.path.path }}"
```
[the remark this answers is not in the document]
Comment (maciaszczykm) [reply modules/web/src/common/components/ingressrulelist/template.html:51]: @marcosdiez Fix it please.

[the remark this answers is not in the document]
Comment (maciaszczykm) [reply modules/web/src/common/components/ingressrulelist/template.html:51]: @marcosdiez It's not ready. We need to fix that issue.

[the remark this answers is not in the document]
Comment (marcosdiez) [reply modules/web/src/common/components/ingressrulelist/template.html]: @maciaszczykm I did fix it. Somehow the github UI did not approve update:

image

**Review by maciaszczykm — COMMENTED** [review]
/lgtm
/approve

**Review by maciaszczykm — CHANGES_REQUESTED** [review]
> There is no need to use `?.` as `*ngIf` above already checks that. I think it should be in the first changed line.

Actually, wait. 

There should be `?.` in the first `*ngIf`: `ingressRuleFlat?.path?.path`. 

Same for last `*ngIf` you've added: `!ingressRuleFlat?.path?.path`.

Additionally, to simplify syntax you could use switch instead of ifs or move the code to component like @floreks suggested.

/approve cancel

**Review by maciaszczykm — APPROVED** [review]

**Pull request merged**